filebeat/module/iis/error/test/iis_error_url.log-expected.json

```diff
@@ -149,7 +157,7 @@
         "source.geo.location.lon": -97.822,
         "source.ip": "149.42.83.135",
         "source.port": 12345,
-        "url.original": "..\\..\\..\\..\\..\\..\\winnt\\win.ini"
+        "url.original": "..\\\\..\\\\..\\\\..\\\\..\\\\..\\\\winnt\\\\win.ini"
```

Comment:
this seems wrong -- I believe that the backslashes in the original event are doubled due to escaping and this seems to escape the escaped backslashes further?

Reply:
I'll take a look on my end as well

Reply:
definitely appears to be the `uri_parts` processor, not the `url_decode`

Reply:
looks like it may be how it reads the field and is doing something like https://stackoverflow.com/a/23363356/15496837

Reply:
it looks like any processor thats reading the string and doing anything with it is being double escaped by Java.  even just 
```
- set:
    field: url.original
    value: "{{_tmp.url_orig}}"
    ignore_failure: true
    if: ctx?._tmp?.url_orig != null && ctx?.url?.original == null
```
is causing the `"url.original": "..\\\\..\\\\..\\\\..\\\\..\\\\..\\\\winnt\\\\win.ini"`

Reply:
https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/ingest/IngestDocument.java#L106 and https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/ingest/IngestDocument.java#L648

Reply:
The templated value `{{_tmp.url_orig}}` will be escaped. That's why `set` now as a `copy_from` option. You can use it instead of a value template and it does an exact string copy with no escaping added.

Reply:
And apparently `{{{foo}}}` (three braces) disables the escaping. https://mustache.github.io/mustache.5.html

That's probably a better way to go in Filebeat since it has better backward compatibility than `copy_from`.

Reply:
ok will try that

Reply:
that worked. Updated.

Reply:
Great, by the way, opened up https://github.com/elastic/elasticsearch/issues/71359 to document the template value interpolation HTML-escaping behavior -- learned something new today 😅

Reply:
Ya definitely something to make sure is documented. :)